src/charm.py

```diff
@@ -198,6 +243,55 @@ def _generate_cu_config(self) -> str:
             tac=self._charm_config.tac,
         )
 
+    @staticmethod
+    def _generate_network_annotations() -> List[NetworkAnnotation]:
+        """Generate a list of NetworkAnnotations to be used by CU's StatefulSet.
+
+        Returns:
+            List[NetworkAnnotation]: List of NetworkAnnotations
+        """
+        return [
+            NetworkAnnotation(
+                name=N3_NETWORK_ATTACHMENT_DEFINITION_NAME,
+                interface=N3_INTERFACE_NAME,
+            )
+        ]
+
+    def _get_nad_base_config(self) -> dict:
+        return {
+            "cniVersion": "0.3.1",
+            "type": "macvlan",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._get_n3_ip_address_from_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+
+    def _network_attachment_definitions_from_config(self) -> list[NetworkAttachmentDefinition]:
+        """Return list of Multus NetworkAttachmentDefinitions to be created based on config."""
+        n3_nad_config = self._get_nad_base_config()
+        if (n3_interface := self._get_n3_interface_from_config()) is not None:
+            n3_nad_config.update({"type": "macvlan", "master": n3_interface})
+        else:
+            n3_nad_config.update({"type": "bridge", "bridge": "ran-br"})
+        return [
+            NetworkAttachmentDefinition(
+                metadata=ObjectMeta(name=N3_NETWORK_ATTACHMENT_DEFINITION_NAME),
+                spec={"config": json.dumps(n3_nad_config)},
+            ),
+        ]
+
+    def _get_n3_interface_from_config(self) -> Optional[str]:
+        return cast(Optional[str], self.model.config.get("n3-interface-name"))
+
+    def _get_n3_ip_address_from_config(self) -> Optional[str]:
+        return cast(Optional[str], self.model.config.get("n3-ip-address"))
```

Comment:
Do we need these? We're using `_charm_config` object everywhere else.

Reply:
Those methods are removed.